### PR TITLE
Bug/1298 wrong update build id

### LINKED ISSUE #1298: Dapp update dialog not showing correct build id
1. build id showing is build id of currently served version, not updated one. 2. build id should be clickable and open github, commit like in the footer.

Of course it makes sense to have 2 not clickable until 1 is fixed

image

### PR DESCRIPTION
**Pull request by ShynRou**
Resolves #1298 

Added a version.json file to public that gets overwritten by the webpack build on production with the correct commit hash.
Used the version.json file to handle the when to show the update dialog.

Also made it possible to make a development build with the build script for testing purposes.

To test the new toast implementation, It's best to run `yarn run build:testnet` and `yarn run serve`, then change the version file in the build directory.

### CODE REVIEW

**Review by tiltom — CHANGES_REQUESTED** [review]

File: scripts/build.js
Comment (ShynRou) [reply scripts/build.js]: Wasn't my change, but removed it for cleanliness.

File: src/app/components/ServiceWorkerToaster/index.tsx
Comment (ShynRou) [reply src/app/components/ServiceWorkerToaster/index.tsx]: Removed it.

File: src/app/components/ServiceWorkerToaster/index.tsx
Comment (ShynRou) [reply src/app/components/ServiceWorkerToaster/index.tsx]: yep just wrapped it and changed it to fetch like the TODO comment requested ^^

File: scripts/build.js
Comment (tiltom) [reply scripts/build.js]: I know it wasn't your change but it looked weird, thanks for removing :)

**Review by tiltom — DISMISSED** [review]

**Review by soulBit — DISMISSED** [review]
Tested locally and all works as expected, approving changes but let's wait for Victor to review before we merge this

**Review by creed-victor — DISMISSED** [review]
Looks good

**Pull request merged**
